comp/dogstatsd/server/rc.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package server implements a component to run the dogstatsd server
+package server
+
+import (
+	"encoding/json"
+	"maps"
+	"slices"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+)
+
+type statsdBlocklistUpdate struct {
+	BlockedMetrics blockedMetrics `json:"blocked_metrics"`
+}
+
+type blockedMetrics struct {
+	ByName byName `json:"by_name"`
+}
+
+type byName struct {
+	Metrics []metricEntry `json:"values"`
+}
+
+type metricEntry struct {
+	Name string `json:"metric_name"`
+}
+
+func (s *server) onBlocklistUpdateCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	s.log.Debug("onBlocklistUpdateCallback received updates:", len(updates))
+
+	// special case: we received a response from RC, but RC didn't have any
+	// configuration for this agent, let's restore the local config and return
+	if len(updates) == 0 {
+		s.restoreBlocklistFromLocalConfig()
```

Comment:
Is this something that can happen normally or more of a bug? It seems to be to be more of the latter, because normally RC expects a callback for each update. In that case, would it make sense to preserve previously set configuration instead of reverting to startup config?

Reply:
I'll confirm this but I'm fairly sure it's normal behaviour, meaning that no configuration has to be applied, intentionally.